tests/kernel/mem_protect/userspace/src/main.c

```diff
@@ -312,21 +321,21 @@ K_APP_DMEM(part0) s32_t size = (0 - CONFIG_PRIVILEGED_STACK_SIZE -
 static void read_priv_stack(void)
 {
 	/* Try to read from privileged stack. */
-#if defined(CONFIG_X86) || defined(CONFIG_ARC)
+#if defined(CONFIG_ARC)
 	int s[1];
 
 	s[0] = 0;
-	priv_stack_ptr = &s[0];
-	priv_stack_ptr = (int *)((unsigned char *)priv_stack_ptr - size);
-#elif defined(CONFIG_ARM)
+	priv_stack_ptr = (char *)&s[0];
+	priv_stack_ptr = (char *)((unsigned char *)priv_stack_ptr - size);
```

Comment:
It would be cleaner to do `priv_stack_ptr = (char *)priv_stack_ptr - size;` now that `priv_stack_ptr` is declared as `char *`.

Reply:
sure I'll fix that